Apps/phazurevmmanagement/microsoftazurevmmanagement_connector.py

```diff
@@ -568,6 +550,107 @@ def _make_rest_call_helper(self, endpoint, action_result, verify=True, headers=N
 
         return phantom.APP_SUCCESS, resp_json
 
+    def _make_redirect_call(self, endpoint, action_result, verify=True, headers=None, params=None, data=None, json=None, method="get"):
+        """ Function that handles instances where a redirect may be required.
+        Default timeout for following the redirect location is 60 seconds.
+
+        :param endpoint: REST endpoint that needs to appended to the service address
+        :param action_result: object of ActionResult class
+        :param headers: request headers
+        :param params: request parameters
+        :param data: request body
+        :param json: JSON object
+        :param method: GET/POST/PUT/DELETE/PATCH (Default will be GET)
+        :param verify: verify server certificate (Default True)
+        :return: status phantom.APP_ERROR/phantom.APP_SUCCESS(along with appropriate message), response, result_location
+        """
+
+        resp_json = None
+        url = "{0}{1}".format(MS_BASE_URL.format(subscriptionId=self._subscription), endpoint)
+        if headers is None:
+            headers = {}
+
+        if not self._access_token:
+            ret_val = self._get_token(action_result)
+
+            if phantom.is_fail(ret_val):
+                return action_result.get_status(), None
+
+        headers.update({
+                'Authorization': 'Bearer {0}'.format(self._access_token),
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'})
+
+        try:
+            request_func = getattr(requests, method)
+        except AttributeError:
+            return action_result.set_status(phantom.APP_ERROR, "Invalid method: {0}".format(method)), resp_json, None
+
+        try:
+            r = request_func(url, json=json, data=data, headers=headers, verify=verify, params=params)
+        except Exception as e:
+            error_msg = self._get_error_message_from_exception(e)
+            return action_result.set_status(phantom.APP_ERROR, "Error Connecting to server. Details: {0}".format(error_msg)), resp_json, None
+
+        if r.text and ('token is invalid' in r.text or 'Access token has expired' in r.text or 'ExpiredAuthenticationToken' in r.text or 'AuthenticationFailed' in r.text):
```

Comment:
since we seem to recheck this same potential string list multiple times in the code can we move them into a constant? and then either make a regex or do something like `any(message in r.text for message in INVALID_TOKEN_MESSAGES)`